WordPress/Sniffs/Theme/NoFaviconSniff.php

```diff
@@ -0,0 +1,76 @@
+<?php
+/**
+ * WordPress Coding Standard.
+ *
+ * @category PHP
+ * @package  PHP_CodeSniffer
+ * @link     https://make.wordpress.org/core/handbook/best-practices/coding-standards/
+ */
+
+/**
+ * WordPress_Sniffs_Theme_NoFaviconSniff.
+ *
+ * ERROR | Verify that no favicon / Apple icon / Windows tile / Android whatever they
+ * call it is being added from the theme. The current check is in Theme-Check plugin is
+ * favicon.php, but could definitely use some fine-tuning and improvement.
+ *
+ * @category  PHP
+ * @package   PHP_CodeSniffer
+ * @author    khacoder
+ */
+class WordPress_Sniffs_Theme_NoFaviconSniff implements PHP_CodeSniffer_Sniff {
+
+	/**
+	 * A list of tokenizers this sniff supports.
+	 *
+	 * @var array
+	 */
+	public $supportedTokenizers = array(
+		'PHP',
+	);
+
+	/**
+	 * Returns an array of tokens this test wants to listen for.
+	 *
+	 * @return array
+	 */
+	public function register() {
+		return array(
+			T_STRING,
+			T_CONSTANT_ENCAPSED_STRING,
```

Comment:
Better to use the build in `PHP_CodeSniffer_Tokens::$stringTokens` and add `T_INLINE_HTML` and any other tokens you really need, than hard-code `T_CONSTANT_ENCAPSED_STRING`.

```php
$tokens   = PHP_CodeSniffer_Tokens::$stringTokens;
$tokens[] = T_INLINE_HTML;
retuen $tokens;
```